lib/chef/provider/aws_route_table.rb

```diff
@@ -31,14 +31,20 @@ def update_aws_object(route_table)
     if new_resource.vpc
       desired_vpc = Chef::Resource::AwsVpc.get_aws_object(new_resource.vpc, resource: new_resource)
       if vpc != desired_vpc
-        raise "VPC of route table #{new_resource.name} (#{route_table.id}) is #{route_table.vpc.id}, but desired vpc is #{new_resource.vpc}!  Moving (or rather, recreating) a route table is not yet supported."
+        # TODO I think we can manage this by deleting the default route_table when switching to a defined one
+        # and then re-creating it when we switch from `main_route_table` to `main_routes`
+        raise "VPC of route table #{new_resource.to_s} is #{route_table.vpc.id}, but desired vpc is #{new_resource.vpc}!  The AWS SDK does not support updating the main route table except by creating a new route table."
       end
     end
   end
 
   def destroy_aws_object(route_table)
-    converge_by "delete route table #{new_resource.name} (#{route_table.id}) in #{region}" do
-      route_table.delete
+    converge_by "delete route table #{new_resource.to_s} in #{region}" do
+      begin
+        route_table.delete
+      rescue AWS::EC2::Errors::DependencyViolation
+        raise "#{new_resource.to_s} could not be deleted because it is the main route table for #{route_table.vpc.id}"
```

Comment:
TODO - test if this is the same error raised when trying to delete a route table in use by a subnet.  If so, change the error to say it is either the main route table for a VPC or is used by a subnet.